### ISSUE TITLE
Two feature requests

### ISSUE THREAD

**Issue by anddis** [chstock/DTComPair#2]
Hello Christian,

Thank you for this very useful package!

If you don't mind, I'd have two feature requests for you to consider:

1. Add a function to compare TPF and TNF on a relative scale [1] (and/or FPF, since relative FPF is estimable also from screen-positive paired data, unlike relative TNF)

2. Calculate and return the estimated covariance $\hat\sigma_{PN}$ [2] (`pv.rpv()` function)

Thank you for considering these requests,
Andrea

[1] Cheng, H. and Macaluso, M., 1997. Comparison of the accuracy of two tests with a confirmatory procedure limited to positive results. Epidemiology, pp.104-106.
 
[2] Moskowitz, C.S. and Pepe, M.S., 2006. Comparing the predictive values of diagnostic tests: sample size and analysis for paired study designs. Clinical trials, 3(3), pp.272-279.

**Comment by chstock** [comment]
Hi Andrea,
many thanks for these valuable suggestions. I actually hadn't been aware of the paper by Cheng and Macaluso. The method fits into the scope of the package and seems worthwhile to implement. My aim with the package currently is merely to maintain CRAN compatibility, due to limited available time time to develop it further. Thus I cannot guarantee your request to be dealt with very soon. However, please also feel free to develop the code, if you like, and make a PR.
Best, Christian

**Comment by anddis** [comment]
Hello Christian,

I forked your DTComPair repository a couple of weeks ago and now I have pushed a few updates to the forked repo:

- `pv.rpv()` now returns the full variance-covariance matrix (`Sigma`);
- `ellipse.pv.rpv()` generates the data to plot a joint confidence region for rPPV and rNPV (depends on the `ellipse` package) (as in Moskowitz and Pepe, 2006);
- `sesp.rel()` calculates relative sensitivity and relative specificity (with Wald CIs and p-value);
- `tpffpf.rel()` calculates relative sensitivity (rTPF) and relative 'one minus specificity' (rFPF) (with Wald CIs and p-value), but it does not calculate their individual components (ie, TPFs and FPFs). This function is meant to be used with paired screen-positive designs, where only rTPF and rFPF are estimable form the data (see Cheng and Macaluso, 1997 or Alonzo, Pepe, Moskowitz, 2002).

Take a look at the functions whenever you feel like doing so and please let me know what you think. Any feedback is welcome. Note that I haven't written any documentation yet, but will be happy to do so if you think that these new functions (not necessarily in their current form, of course) can be of value for the `DTComPair` package.

Thank you again for such a useful package!
Andrea

```
# Example

data(Paired1)  
ftable(Paired1)
paired.layout <- tab.paired(d=d, y1=y1, y2=y2, data=Paired1)
paired.layout 
rpv.results <- pv.rpv(paired.layout, alpha = 0.05)
rpv.results
sesp.rel(paired.layout, alpha = 0.05)
tpffpf.rel(paired.layout, alpha = 0.05) # meant for paired screen-positive designs
ellipse.data <- ellipse.pv.rpv(rpv.results, exponentiate = FALSE)
plot(ellipse.data$ellipse, type = "l", ylim = log(c(0.8, 1.5)), xlim = log(c(0.8, 1.2)))
points(ellipse.data$centre[1], ellipse.data$centre[2], col = "red", pch = 19)
abline(h = 0, v = 0, lty = 3)
```

--

Alonzo, T.A., Pepe, M.S. and Moskowitz, C.S., 2002. Sample size calculations for comparative studies of medical tests for detecting presence of disease. Statistics in medicine, 21(6), pp.835-852.

**Comment by chstock** [comment]
Hi Andrea,

thanks for this nice addition and the new incomplete 'gold standard' data perspective, this is really appreciated. I had a look at the code and the papers, and ran the example. I'd be happy to incorporate the functions and documentation into the package and make a CRAN update. 

With respect to `tpffpf.rel()`, it would be great if an example in the documentation could reproduce results in any of the papers, but this is only a suggestion. Also, it would be great if the documentation could give a hint (suggested wording) on the interpretation of the new relative metrics.

The present documentation, while still working, is not (yet) done according to current standards, e.g. as suggested by roxygen2 and other packages. In case you wonder and would like to do it in a more contemporary fashion, this fully fine, and even appreciated.

Please indicate whether you would like to become author or contributor (both is justified, I think). You could also directly amend the DESCRIPTION file.

Let me know if you have questions.

**Comment by anddis** [comment]
Hello Christian,

Thanks for your reply.

In terms of documentation, I:

- documented `Sigma` (under Values) in the existing `pv.rpv.Rd` file
- documented `ellipse.pv.rpv` (using roxygen2)
- documented `sesp.rel` (using roxygen2)
- documented `tpffpf.rel` (using roxygen2). I used example data from Cheng and Macaluso, as you suggested.

In the help files for `sesp.rel` and `tpffpf.rel`, I added some language about the interpretation of those measures, as you suggested (under Details). Is this what you had in mind?

Let me know if you have any feedback regarding the documentation.

Other points: 

- one could consider using the generic function `ellipse::ellipse`, but this would require introducing a `pv.rpv` class. I don't think this is particularly important, but perhaps something you might want to consider in the future?

- thank you, I appreciate very much your offer about co-authorship, which I will happily accept. I have modified the `DESCRIPTION` and `CITATION` files. Let me know whether my edits are ok by you.

Cheers,
Andrea
